src/System.Memory/tests/ParsersAndFormatters/Parser/TestData.Parser.Date.cs

```diff
@@ -12,7 +13,32 @@ public static IEnumerable<ParserTestData<DateTime>> DateTimeParserTestData
         {
             get
             {
-                foreach (ParserTestData<DateTime> testData in DateTimeFormatterTestData.ToParserTheoryDataCollection())
+				// Format J tests for DateTime parsing.
+				yield return new ParserTestData<DateTime>("0997-07-16", DateTime.Parse("0997-07-16"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16", DateTime.Parse("1997-07-16"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20", DateTime.Parse("1997-07-16T19:20"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30", DateTime.Parse("1997-07-16T19:20:30"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.45", DateTime.Parse("1997-07-16T19:20:30.45"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.4555555", DateTime.Parse("1997-07-16T19:20:30.4555555"), 'J', expectedSuccess: true);
+                
+                // Test fraction rounding.
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.45555554", DateTime.Parse("1997-07-16T19:20:30.45555554"), 'J', expectedSuccess: true);
```

Comment:
The two input strings don't match here. Was that intentional?

Reply:
This applies to the next line.
Yes, it's intentional. We expect the parser to truncate, but `DateTime.Parse` will round up in this case which will return a different result. So, we pass the expected DateTime string to `DateTime.Parse`.

Reply:
In that case. maybe we should update `DateTime.Parse("1997-07-16T19:20:30.45555554")` above as well to only have 7 digits (i.e. remove the last digit 4). Because that test case makes it look like the behavior is consistent (both round, when in reality one is truncating).